script/tasks/diamondSyncWhitelist.sh

```diff
@@ -0,0 +1,739 @@
+          else
+            # No pairs found using fallback method
+            :
+          fi
+        else
+          # getWhitelistedAddresses also failed
+          :
+        fi
+
+        # Attempt $ATTEMPT failed, waiting 3 seconds before retry...
+        sleep 3
+        ATTEMPT=$((ATTEMPT + 1))
+      done
+
+      # All attempts failed to get whitelisted pairs
+      return 1
+    }
+
+    # Get required contract-selector pairs from whitelist files
+    echoDebug "Getting required contract-selector pairs from whitelist files for network: $NETWORK"
+    REQUIRED_PAIRS=($(getContractSelectorPairs "$NETWORK"))
+    
+    echoDebug "Found ${#REQUIRED_PAIRS[@]} required pairs from whitelist files"
+    if [[ ${#REQUIRED_PAIRS[@]} -gt 0 ]]; then
+      echoDebug "Required pairs: ${REQUIRED_PAIRS[@]}"
+    fi
+    
+    if [[ ${#REQUIRED_PAIRS[@]} -eq 0 ]]; then
+      printf '\033[0;33m%s\033[0m\n' "⚠️  [$NETWORK] No contract-selector pairs found in whitelist files for this network"
+      return
+    fi
+
+    # Get current whitelisted pairs from diamond
+    CURRENT_PAIRS=($(getCurrentWhitelistedPairs))
+    local get_pairs_exit_code=$?
+    
+    if [[ $get_pairs_exit_code -ne 0 ]]; then
+      printf '\033[0;31m%s\033[0m\n' "❌ [$NETWORK] Unable to fetch current whitelisted pairs"
+      {
+        echo "[$NETWORK] Error: Unable to fetch current whitelisted pairs"
+        echo ""
+      } >> "$FAILED_LOG_FILE"
+      return
+    fi
+
+    # Determine missing pairs
+    NEW_PAIRS=()
+    NEW_ADDRESSES=()
+    
+    for REQUIRED_PAIR in "${REQUIRED_PAIRS[@]}"; do
+      # Split the pair by '|' character using parameter expansion
+      ADDRESS="${REQUIRED_PAIR%%|*}"
+      SELECTORS_STR="${REQUIRED_PAIR#*|}"
+      
+      # Check for address zero (forbidden)
+      # Handle all variations: uppercase, lowercase, mixed case
+      ADDRESS_NORMALIZED=$(echo "$ADDRESS" | tr '[:upper:]' '[:lower:]' | sed 's/^0x//')
+      if [[ "$ADDRESS_NORMALIZED" == "0000000000000000000000000000000000000000" ]]; then
+        printf '\033[0;31m%s\033[0m\n' "❌ [$NETWORK] Error: Whitelisting address zero is forbidden: $ADDRESS"
+        printf '\033[0;33m%s\033[0m\n' "⚠️  Please check whitelist.json or whitelist.staging.json"
+        printf '\033[0;33m%s\033[0m\n' "⚠️  Remove address zero from configuration and discuss with backend team"
+        {
+          echo "[$NETWORK] Error: Whitelisting address zero is forbidden: $ADDRESS"
+          echo "[$NETWORK] This address is invalid and should not be in the whitelist configuration"
+          echo "[$NETWORK] Please check whitelist.json or whitelist.staging.json and remove address zero"
+          echo "[$NETWORK] Discuss with backend team if this address is needed"
+          echo ""
+        } >> "$FAILED_LOG_FILE"
+        return 1
+      fi
+      
+      # Check if address has code
+      CHECKSUMMED=$(cast --to-checksum-address "$ADDRESS")
+      CODE=$(cast code "$CHECKSUMMED" --rpc-url "$RPC_URL")
```

Comment:
pls make sure that we run this check only on addresses that will be newly added and not on every address (that is already whitelisted). 

Also, we can consider parallelizing these calls for faster script runtime.

But maybe that is an improvement for later, we should focus now on getting this topic out of the door.

Reply:
good point, created ticket for it: https://lifi.atlassian.net/browse/LF-16406